state/errors.go

```diff
@@ -10,19 +10,23 @@ import (
 
 // ErrMissingTrie is an error-compatible struct holding the root hash of the trie that is missing
 type ErrMissingTrie struct {
-	rootHash []byte
+	rootHash    []byte
+	originalErr error
 }
 
 // ------- ErrMissingTrie
 
 // NewErrMissingTrie returns a new instantiated struct
-func NewErrMissingTrie(rootHash []byte) *ErrMissingTrie {
-	return &ErrMissingTrie{rootHash: rootHash}
+func NewErrMissingTrie(rootHash []byte, originalErr error) *ErrMissingTrie {
+	return &ErrMissingTrie{
+		rootHash:    rootHash,
+		originalErr: originalErr,
+	}
 }
 
 // Error returns the error as string
 func (e *ErrMissingTrie) Error() string {
-	return "trie was not found for hash " + hex.EncodeToString(e.rootHash)
+	return fmt.Sprintf("trie was not found for hash, rootHash = %s, err = %s", hex.EncodeToString(e.rootHash), e.originalErr.Error())
```

Comment:
since you don't check `originalErr` for `nil` at the constructor level, this line can cause the code to panic. might check on the constructor for nil errors

Reply:
Fixed.